UI/MiscScreens.cpp

```diff
@@ -707,7 +707,7 @@ void LogoScreen::sendMessage(const char *message, const char *value) {
 }
 
 bool LogoScreen::key(const KeyInput &key) {
-	if (key.deviceId != DEVICE_ID_MOUSE) {
+	if (key.deviceId != DEVICE_ID_MOUSE && key.flags & KEY_DOWN) {
```

Comment:
This should be:

`if (key.deviceId != DEVICE_ID_MOUSE && (key.flags & KEY_DOWN))`

Yes you do need those extra parenthesis in C++...

Reply:
Bitwise have precedence over logical: https://en.cppreference.com/w/cpp/language/operator_precedence. It should be fine like this no?

Reply:
I don't know how you read the table, but logical have precedence over bitwise, not the other way around.

Reply:
In the table the operator with the smaller number get evaluated before, if two operator have the same level of precedence they get evaluated based on the associativity column (i.e. left-to-right or right-to-left).

I double checked with the compiler as well:

```
printf("Default: %d, &&: %d, &: %d\n", true && 0x10 & 0x01, (true && 0x10) & 0x01, true && (0x10 & 0x01));
// Result: Default: 0, &&: 1, &: 0
```

Anyway this is for sure a source of confusion so better use parenthesis (you might be confusing with & and == that have this need mb?).

Reply:
Oh duh, yeah, I saw the != and thought that would be involved in it, but you're right - in this case it'll work. 

However, I really do want parenthesis for this, because this kind of thing has been the source of many bugs.